src/sign-in-xplat/MainPage.xaml.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Microsoft.Graph;
+using Microsoft.Identity.Client;
+using Microsoft.Maui.Controls;
+using Microsoft.Maui.Essentials;
+
+namespace XPlat
+{
+    public partial class MainPage : ContentPage
+    {
+        //Set the scope for API call to user.read
+        private string[] scopes = new string[] { "user.read" };
+        private const string ClientId = "[Enter the Client Id (Application ID obtained from the Azure portal), e.g. ba74781c2-53c2-442a-97c2-3d60re42f403]";
+        private const string Tenant = "[Copy the Tenant Id from the Azure portal]";
+        private const string Authority = "https://login.microsoftonline.com/" + Tenant;
+
+        // The MSAL Public client app
+        private static IPublicClientApplication PublicClientApp;
+
+        private static string MSGraphURL = "https://graph.microsoft.com/v1.0/";
+        private static AuthenticationResult authResult;
+
+        public MainPage()
+        {
+            InitializeComponent();
+        }
+
+        private async void OnSignInClicked(object sender, EventArgs e)
+        {
+            try
+            {
+                // Sign-in user using MSAL and obtain an access token for MS Graph
+                GraphServiceClient graphClient = await SignInAndInitializeGraphServiceClient(scopes);
+
+                // Call the /me endpoint of Graph
+                User graphUser = await graphClient.Me.Request().GetAsync();

```

Comment:
I'm going to make the suggestion before Marsh does (and this is based off of the recent feedback on the python webapp work), instead of hiding this in a helper method, please add it all inline here.  Specifically to help the user see (without jumping context) what code they are responsible for.  I had to do the same with the Python app code, creating very "fat" (controller) methods just to be very clear about the "flow" of things.  What you have here is closer to what we would normally write day-to-day, but not what we want for the tutorial series, where possible.

Reply:
thanks for the feedback @ckittel, please let's confirm another level of inlining is desired in here. I understand the concept of having the code handy but at the same time I am worried about its readability.

partially done | addressed from [0c098](https://github.com/Azure-Samples/ms-identity-docs-code-csharp/pull/7/commits/0c098c645c61917802a67d423b30d3f2790c0708)

Reply:
Good progress, like you said only partially, the next step is probably the most important, and that's getting the `SignInUserAndGetTokenUsingMSAL` bits inlined -- as that's the core of what we're teaching here, and with proper comments above every "step" along the way, it'll be very clear about what the logical flow of the process is.

Reply:
a bit more progress on the in-lining. We inlined the `PublicClientApp` instantiation. 

Please let me know if we want to inline the fill `SignInUserAndGetTokenUsingMSAL` method.

Reply:
Yeah, I think getting `SignInUserAndGetTokenUsingMSAL` brought into the function body here will be helpful for content development and fulfilling the "all in one place" request that Marsh has of the MSAL work being done.  After that, I'm seeing no "blockers" here for merge.

Reply:
sure thing @ckittel, all-in-lined now!


done | addressed from [`c294d`](https://github.com/Azure-Samples/ms-identity-docs-code-csharp/pull/11/commits/c294d7b3f6c0d0ee06f4a515d48236cb30029b17)

Reply:
Thanks, looks like what we've been instructed to do. 💪